framework/src/postprocessors/PseudoTimestep.C

```diff
@@ -0,0 +1,190 @@
+  return res_norm;
+}
+
+void
+PseudoTimestep::output_pseudo_timestep(Real & curr_dt, const ExecFlagType & exec_type)
+{
+  unsigned int curr_step = _fe_problem.timeStep();
+
+  unsigned int steps_size = _iterations_step_sequence.size();
+  unsigned int res_size = _residual_norms_sequence.size();
+
+  Moose::out << std::endl;
+  Moose::out << "***********************************************" << std::endl;
+  Moose::out << "Current step " << curr_step << " executed at " << _current_execute_flag
+             << " current timestep " << curr_dt << std::endl;
+
+  Moose::out << "Iterations vector is of size " << steps_size << " , and residual norms is of size "
+             << res_size << std::endl;
+  Moose::out << "Current dt: " << _iterations_step_sequence[steps_size - 1]
+             << " and residual: " << _residual_norms_sequence[res_size - 1] << std::endl;
+
+  Moose::out << std::endl;
+}
+
+Real
+PseudoTimestep::timestep_SER()
+{
+  unsigned int curr_step = _fe_problem.timeStep();
+
+  unsigned int steps_size = _iterations_step_sequence.size();
+  unsigned int res_size = _residual_norms_sequence.size();
+  unsigned int prev_steps = std::min(_iterations_window, curr_step);
+
+  Real factor = std::pow(_residual_norms_sequence[res_size - prev_steps] /
+                             _residual_norms_sequence[res_size - 1],
+                         _alpha);
+
+  Real update_dt = _iterations_step_sequence[steps_size - 1] * factor;
+  return update_dt;
+}
+
+Real
+PseudoTimestep::timestep_RDM()
+{
+  unsigned int steps_size = _iterations_step_sequence.size();
+  unsigned int res_size = _residual_norms_sequence.size();
+
+  Real factor = (_residual_norms_sequence[res_size - 2] - _residual_norms_sequence[res_size - 1]) /
+                _residual_norms_sequence[res_size - 2];
+
+  Real update_dt = _iterations_step_sequence[steps_size - 1] * std::pow(_alpha, factor);
+  return update_dt;
+}
+
+Real
+PseudoTimestep::timestep_EXP()
+{
+  Real factor = std::pow(_alpha, _fe_problem.timeStep() - 1);
+
+  Real update_dt = _initial_dt * factor;
+  return update_dt;
+}
+
+void
+PseudoTimestep::execute()
+{
+  Transient * transient = dynamic_cast<Transient *>(_app.getExecutioner());
+
+  Real res_norm, var_res;
+  Real curr_dt;
+  Real update_dt;
+
+  if (_current_execute_flag == EXEC_INITIAL)
+  {
+    _dt = _initial_dt;
+  }
+
+  if (_current_execute_flag == EXEC_TIMESTEP_END)
+  {
+    res_norm = current_residual_norm();
+    _residual_norms_sequence.push_back(res_norm);
+
+    curr_dt = transient->getDT();
+    _iterations_step_sequence.push_back(curr_dt);
+
+    _dt = curr_dt;
+    if (_fe_problem.timeStep() > 1)
+    {
+      update_dt = curr_dt;
+      switch (_method)
+      {
+        case 1:
+          update_dt = timestep_SER();
+          break;
+        case 2:
+          update_dt = timestep_EXP();
+          break;
+        case 3:
+          update_dt = timestep_RDM();
+          break;
+      }
+      _dt = std::min(_max_dt, update_dt);
```

Comment:
```suggestion
  if (_current_execute_flag == EXEC_INITIAL)
  {
    _dt = _initial_dt;
  }

  if (_current_execute_flag == EXEC_TIMESTEP_END)
  {
    Real res_norm = current_residual_norm();
    _residual_norms_sequence.push_back(res_norm);

    Real curr_dt = transient->getDT();
    _iterations_step_sequence.push_back(curr_dt);

    _dt = curr_dt;
    if (_fe_problem.timeStep() > 1)
    {
      Real update_dt = curr_dt;
      switch (_method)
      {
        case TimestepperMethod::SER:
          update_dt = timestep_SER();
          break;
        case TimestepperMethod::EXP:
          update_dt = timestep_EXP();
          break;
        case TimestepperMethod::RDM:
          update_dt = timestep_RDM();
          break;
      }
      _dt = std::min(_max_dt, update_dt);
```

Reply:
Is this only running clang format? Cause that's what I was letting CIVET do it for a draft.

Reply:
@lynnmunday I still don't know what was the problem here. It's outdated but not by much.

Reply:
I was suggesting to use an enum class for the switch statement instead of numbers.  I think @grmnptr in the below comment is suggesting to stay with the MooseEnum.